lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -333,31 +447,40 @@ def delete(self, username):
         """
         # get the auth token
         auth_header = request.headers.get("Authorization")
-        auth_token = ""
-        if auth_header:
-            try:
-                auth_token = auth_header.split(" ")[1]
-            except IndexError:
-                self.app.logger.info(
-                    f"failed to delete the user due to malformed Bearer token: {auth_header}"
-                )
-                abort(401, message="Bearer token malformed, failed to delete the user")
 
-        if auth_token:
+        if not auth_header:
+            self.app.logger.error(f"Auth token not provided while deleting a user")
+            abort(
+                400,
+                message="Please add 'Authorization' token as Authorization: Bearer <JWT_Auth_token>",
+            )
+
+        try:
+            auth_token = auth_header.split(" ")[1]
+        except IndexError:
+            self.app.logger.error(f"Malformed Auth header while deleting a user")
+            abort(
+                400,
+                message="Malformed Auth header, please add 'Authorization' token as Authorization: Bearer <JWT_Auth_token>",
+            )
+
+        try:
             user_id = UserModel.decode_auth_token(auth_token, self.app)
-            user = UserModel.query.filter_by(username=username).first()
-            if user.id != user_id:
-                self.app.logger.info(
-                    f"User_id decoded from the auth token {user_id} is different from the one provided in the URI {user_id}"
-                )
-                abort(
-                    403,
-                    message="Authentication token does not belong to the user, failed to delete the user",
-                )
 
             if not isinstance(user_id, str):
-                UserModel.query.filter_by(id=user_id).delete()
+                q = UserModel.query.filter_by(id=user_id)
+                if q.first().username != username:
+                    self.app.logger.info(
+                        f"Username retrieved from the auth token {user_id} is different for the username provided while deleting user entry"
```

Comment:
This needs to be more nuanced. I'm not sure we want to code privileges into the user model in any detail, but an Admin user should be able to delete other users.

A simple way to start would be either a `user.isPrivileged()` boolean method that's currently always `False` or even does `user.username == "Admin"`. Or we could consider adding an `isAdmin` boolean column in the DB that we set to `True` for the (probably pre-defined) `Admin` user and will initially be `False` for everyone else.

Reply:
I think having a column in the db whose value gonna be always false except for couple user would not bring much benefits, I think having a unique `Admin/admin` username would do fine. thoughts?

Reply:
I suspect at some point we'll want groups and probably a permissions scheme. The question is how far we want to push towards prototyping before we know exactly what we want.

I threw out the "admin" column idea because after the first step of a built-in pre-defined "Admin" user, this would let us designate "portante" as an admin.

But while that's something to think about now, I think that an explicit `username == "Admin"` would be fine for your first pass here. I do think we should hide this behind an API so we can easily re-implement in one place when we're ready.